tensorboard/webapp/metrics/views/main_view/common_selectors_test.ts

```diff
@@ -430,4 +552,351 @@ describe('common selectors', () => {
+            maxValue: 10,
+            filterLowerValue: 1,
+            filterUpperValue: 5,
+            includeUndefined: true,
+          },
+          undefined
+        )
+      ).toBeTrue();
+
+      expect(
+        selectors.TEST_ONLY.utils.matchFilter(
+          {
+            type: DomainType.INTERVAL,
+            minValue: 0,
+            maxValue: 10,
+            filterLowerValue: 1,
+            filterUpperValue: 5,
+            includeUndefined: false,
+          },
+          undefined
+        )
+      ).toBeFalse();
+    });
+
+    it('returns values including value when filter type is DISCRETE', () => {
+      const filter: DiscreteFilter = {
+        type: DomainType.DISCRETE,
+        includeUndefined: false,
+        possibleValues: [],
+        filterValues: ['afoo', 'foob', 'foo', 'fo'],
+      };
+      expect(selectors.TEST_ONLY.utils.matchFilter(filter, 'foo')).toBeTrue();
+      expect(selectors.TEST_ONLY.utils.matchFilter(filter, 'bar')).toBeFalse();
+    });
+
+    it('checks if value is within bounds when filter type is INTERVAL', () => {
+      const filter: IntervalFilter = {
+        type: DomainType.INTERVAL,
+        includeUndefined: true,
+        minValue: 0,
+        maxValue: 10,
+        filterLowerValue: 1,
+        filterUpperValue: 5,
+      };
+      expect(selectors.TEST_ONLY.utils.matchFilter(filter, 0)).toBeFalse();
+      expect(selectors.TEST_ONLY.utils.matchFilter(filter, 1)).toBeTrue();
+      expect(selectors.TEST_ONLY.utils.matchFilter(filter, 3)).toBeTrue();
+      expect(selectors.TEST_ONLY.utils.matchFilter(filter, 5)).toBeTrue();
+      expect(selectors.TEST_ONLY.utils.matchFilter(filter, 6)).toBeFalse();
+
+      expect(selectors.TEST_ONLY.utils.matchFilter(filter, 'foo')).toBeFalse();
+    });
+
+    it('returns false if filter type is neither DISCRETE nor INTERVAL', () => {
+      expect(
+        selectors.TEST_ONLY.utils.matchFilter({} as DiscreteFilter, 'foo')
+      ).toBeFalse();
+    });
+  });
+
+  describe('filterRunItemsByRegex', () => {
+    it('returns all runs when no regex is provided', () => {
+      expect(
+        selectors.TEST_ONLY.utils.filterRunItemsByRegex(
+          runTableItems,
+          '',
+          false
+        )
+      ).toEqual(runTableItems);
+    });
+
+    it('only returns runs matching regex', () => {
+      expect(
+        selectors.TEST_ONLY.utils.filterRunItemsByRegex(
+          runTableItems,
+          'run',
+          false
+        )
+      ).toEqual(runTableItems);
+
+      expect(
+        selectors.TEST_ONLY.utils.filterRunItemsByRegex(
+          runTableItems,
+          'exp',
+          false
+        )
+      ).toEqual([]);
+
+      expect(
+        selectors.TEST_ONLY.utils.filterRunItemsByRegex(
+          runTableItems,
+          'exp',
+          true
+        )
+      ).toEqual(runTableItems);
+
+      expect(
+        selectors.TEST_ONLY.utils.filterRunItemsByRegex(
+          runTableItems,
+          'run1',
```

Comment:
If you make the 3rd run run3 I think it will serve this test well to make it 'run[13]'.